{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "a133a6c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import networkx as nx\n",
    "import csv\n",
    "from xml.dom import minidom\n",
    "import xml.etree.ElementTree as ET\n",
    "from xml.etree.ElementTree import tostring\n",
    "import numpy as np # linear algebra\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c0d3cd88-6213-4ca6-9bb4-5e438e723d4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import networkx as nx\n",
    "import csv\n",
    "from xml.dom import minidom\n",
    "import xml.etree.ElementTree as ET\n",
    "from xml.etree.ElementTree import tostring\n",
    "import numpy as np # linear algebra\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d4de8f86",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FileName</th>\n",
       "      <th>GenomeSize</th>\n",
       "      <th>GeneSize</th>\n",
       "      <th>Coding</th>\n",
       "      <th>ASR</th>\n",
       "      <th>assembly_accession</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>total_gene_count</th>\n",
       "      <th>protein_coding_gene_count</th>\n",
       "      <th>...</th>\n",
       "      <th>Length_mean_exons</th>\n",
       "      <th>Length_count_introns</th>\n",
       "      <th>Length_mean_introns</th>\n",
       "      <th>Num_proteins</th>\n",
       "      <th>Num_transcripts</th>\n",
       "      <th>Num_proteins_hsapiens</th>\n",
       "      <th>Num_transcripts_hsapiens</th>\n",
       "      <th>Num_rnaseq_reads</th>\n",
       "      <th>Num_tissues</th>\n",
       "      <th>Num_runs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000001635.27_GRCm39_genomic.gff</td>\n",
       "      <td>2728222451</td>\n",
       "      <td>1320613304</td>\n",
       "      <td>37246699</td>\n",
       "      <td>5.327877</td>\n",
       "      <td>GCF_000001635.27</td>\n",
       "      <td>Mus musculus</td>\n",
       "      <td>GRCm39</td>\n",
       "      <td>50763</td>\n",
       "      <td>22198</td>\n",
       "      <td>...</td>\n",
       "      <td>482.807</td>\n",
       "      <td>295831</td>\n",
       "      <td>6519.53</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_000003025.6_Sscrofa11.1_genomic.gff</td>\n",
       "      <td>2501912388</td>\n",
       "      <td>1389570318</td>\n",
       "      <td>35553489</td>\n",
       "      <td>3.703144</td>\n",
       "      <td>GCF_000003025.6</td>\n",
       "      <td>Sus scrofa</td>\n",
       "      <td>Sscrofa11.1</td>\n",
       "      <td>30371</td>\n",
       "      <td>20666</td>\n",
       "      <td>...</td>\n",
       "      <td>439.112</td>\n",
       "      <td>240989</td>\n",
       "      <td>7640.43</td>\n",
       "      <td>58148</td>\n",
       "      <td>3619303</td>\n",
       "      <td>47060</td>\n",
       "      <td>0</td>\n",
       "      <td>12069331098</td>\n",
       "      <td>42</td>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_000165445.2_Mmur_3.0_genomic.gff</td>\n",
       "      <td>2487409138</td>\n",
       "      <td>1250699459</td>\n",
       "      <td>35131539</td>\n",
       "      <td>3.540079</td>\n",
       "      <td>GCF_000165445.2</td>\n",
       "      <td>Microcebus murinus</td>\n",
       "      <td>Mmur_3.0</td>\n",
       "      <td>31964</td>\n",
       "      <td>20522</td>\n",
       "      <td>...</td>\n",
       "      <td>344.445</td>\n",
       "      <td>245301</td>\n",
       "      <td>6754.20</td>\n",
       "      <td>208276</td>\n",
       "      <td>8993166</td>\n",
       "      <td>174117</td>\n",
       "      <td>8993046</td>\n",
       "      <td>3364726994</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_000247795.1_Bos_indicus_1.0_genomic.gff</td>\n",
       "      <td>2673965444</td>\n",
       "      <td>987855691</td>\n",
       "      <td>32410319</td>\n",
       "      <td>2.008855</td>\n",
       "      <td>GCF_000247795.1</td>\n",
       "      <td>Bos indicus</td>\n",
       "      <td>Bos_indicus_1.0</td>\n",
       "      <td>26229</td>\n",
       "      <td>19786</td>\n",
       "      <td>...</td>\n",
       "      <td>308.562</td>\n",
       "      <td>193564</td>\n",
       "      <td>5542.66</td>\n",
       "      <td>81967</td>\n",
       "      <td>1989647</td>\n",
       "      <td>45075</td>\n",
       "      <td>344947</td>\n",
       "      <td>8324786874</td>\n",
       "      <td>198</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000317375.1_MicOch1.0_genomic.gff</td>\n",
       "      <td>2287340943</td>\n",
       "      <td>943175664</td>\n",
       "      <td>33754777</td>\n",
       "      <td>2.229472</td>\n",
       "      <td>GCF_000317375.1</td>\n",
       "      <td>Microtus ochrogaster</td>\n",
       "      <td>MicOch1.0</td>\n",
       "      <td>26397</td>\n",
       "      <td>20077</td>\n",
       "      <td>...</td>\n",
       "      <td>279.875</td>\n",
       "      <td>197625</td>\n",
       "      <td>5402.20</td>\n",
       "      <td>104813</td>\n",
       "      <td>8780884</td>\n",
       "      <td>51802</td>\n",
       "      <td>0</td>\n",
       "      <td>670954419</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>GCF_002201575.2_ASM220157v2_genomic.gff</td>\n",
       "      <td>2419443398</td>\n",
       "      <td>929442675</td>\n",
       "      <td>32966902</td>\n",
       "      <td>1.805260</td>\n",
       "      <td>GCF_002201575.2</td>\n",
       "      <td>Neomonachus schauinslandi</td>\n",
       "      <td>ASM220157v2</td>\n",
       "      <td>27736</td>\n",
       "      <td>19838</td>\n",
       "      <td>...</td>\n",
       "      <td>234.859</td>\n",
       "      <td>191095</td>\n",
       "      <td>5240.15</td>\n",
       "      <td>71161</td>\n",
       "      <td>758492</td>\n",
       "      <td>62812</td>\n",
       "      <td>0</td>\n",
       "      <td>403656856</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>GCF_002263795.3_ARS-UCD2.0_genomic.gff</td>\n",
       "      <td>2770686120</td>\n",
       "      <td>1295081857</td>\n",
       "      <td>36598587</td>\n",
       "      <td>3.620578</td>\n",
       "      <td>GCF_002263795.3</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>ARS-UCD2.0</td>\n",
       "      <td>37073</td>\n",
       "      <td>21667</td>\n",
       "      <td>...</td>\n",
       "      <td>482.419</td>\n",
       "      <td>244314</td>\n",
       "      <td>6987.83</td>\n",
       "      <td>98483</td>\n",
       "      <td>3319583</td>\n",
       "      <td>67352</td>\n",
       "      <td>0</td>\n",
       "      <td>8524329442</td>\n",
       "      <td>95</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>GCF_002837175.3_ASM283717v5_genomic.gff</td>\n",
       "      <td>2496225537</td>\n",
       "      <td>1128102550</td>\n",
       "      <td>33823680</td>\n",
       "      <td>2.906335</td>\n",
       "      <td>GCF_002837175.3</td>\n",
       "      <td>Physeter catodon</td>\n",
       "      <td>ASM283717v5</td>\n",
       "      <td>29662</td>\n",
       "      <td>20411</td>\n",
       "      <td>...</td>\n",
       "      <td>314.058</td>\n",
       "      <td>224230</td>\n",
       "      <td>6554.51</td>\n",
       "      <td>254677</td>\n",
       "      <td>100392</td>\n",
       "      <td>166881</td>\n",
       "      <td>0</td>\n",
       "      <td>1697796066</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>GCF_002863925.1_EquCab3.0_genomic.gff</td>\n",
       "      <td>2506966135</td>\n",
       "      <td>1190788025</td>\n",
       "      <td>36221238</td>\n",
       "      <td>3.716273</td>\n",
       "      <td>GCF_002863925.1</td>\n",
       "      <td>Equus caballus</td>\n",
       "      <td>EquCab3.0</td>\n",
       "      <td>33687</td>\n",
       "      <td>20955</td>\n",
       "      <td>...</td>\n",
       "      <td>416.092</td>\n",
       "      <td>250204</td>\n",
       "      <td>6094.25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>GCF_003254725.2_ASM325472v2_genomic.gff</td>\n",
       "      <td>2349862946</td>\n",
       "      <td>1395147252</td>\n",
       "      <td>35822391</td>\n",
       "      <td>4.217887</td>\n",
       "      <td>GCF_003254725.2</td>\n",
       "      <td>Canis lupus dingo</td>\n",
       "      <td>ASM325472v2</td>\n",
       "      <td>40364</td>\n",
       "      <td>20775</td>\n",
       "      <td>...</td>\n",
       "      <td>388.097</td>\n",
       "      <td>283465</td>\n",
       "      <td>7298.66</td>\n",
       "      <td>65282</td>\n",
       "      <td>0</td>\n",
       "      <td>65282</td>\n",
       "      <td>0</td>\n",
       "      <td>9134207560</td>\n",
       "      <td>37</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows Ã— 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       FileName  GenomeSize    GeneSize  \\\n",
       "0           GCF_000001635.27_GRCm39_genomic.gff  2728222451  1320613304   \n",
       "1       GCF_000003025.6_Sscrofa11.1_genomic.gff  2501912388  1389570318   \n",
       "2          GCF_000165445.2_Mmur_3.0_genomic.gff  2487409138  1250699459   \n",
       "3   GCF_000247795.1_Bos_indicus_1.0_genomic.gff  2673965444   987855691   \n",
       "4         GCF_000317375.1_MicOch1.0_genomic.gff  2287340943   943175664   \n",
       "7       GCF_002201575.2_ASM220157v2_genomic.gff  2419443398   929442675   \n",
       "8        GCF_002263795.3_ARS-UCD2.0_genomic.gff  2770686120  1295081857   \n",
       "10      GCF_002837175.3_ASM283717v5_genomic.gff  2496225537  1128102550   \n",
       "11        GCF_002863925.1_EquCab3.0_genomic.gff  2506966135  1190788025   \n",
       "12      GCF_003254725.2_ASM325472v2_genomic.gff  2349862946  1395147252   \n",
       "\n",
       "      Coding       ASR assembly_accession              organism_name  \\\n",
       "0   37246699  5.327877   GCF_000001635.27               Mus musculus   \n",
       "1   35553489  3.703144    GCF_000003025.6                 Sus scrofa   \n",
       "2   35131539  3.540079    GCF_000165445.2         Microcebus murinus   \n",
       "3   32410319  2.008855    GCF_000247795.1                Bos indicus   \n",
       "4   33754777  2.229472    GCF_000317375.1       Microtus ochrogaster   \n",
       "7   32966902  1.805260    GCF_002201575.2  Neomonachus schauinslandi   \n",
       "8   36598587  3.620578    GCF_002263795.3                 Bos taurus   \n",
       "10  33823680  2.906335    GCF_002837175.3           Physeter catodon   \n",
       "11  36221238  3.716273    GCF_002863925.1             Equus caballus   \n",
       "12  35822391  4.217887    GCF_003254725.2          Canis lupus dingo   \n",
       "\n",
       "           asm_name  total_gene_count  protein_coding_gene_count  ...  \\\n",
       "0            GRCm39             50763                      22198  ...   \n",
       "1       Sscrofa11.1             30371                      20666  ...   \n",
       "2          Mmur_3.0             31964                      20522  ...   \n",
       "3   Bos_indicus_1.0             26229                      19786  ...   \n",
       "4         MicOch1.0             26397                      20077  ...   \n",
       "7       ASM220157v2             27736                      19838  ...   \n",
       "8        ARS-UCD2.0             37073                      21667  ...   \n",
       "10      ASM283717v5             29662                      20411  ...   \n",
       "11        EquCab3.0             33687                      20955  ...   \n",
       "12      ASM325472v2             40364                      20775  ...   \n",
       "\n",
       "    Length_mean_exons Length_count_introns  Length_mean_introns  Num_proteins  \\\n",
       "0             482.807               295831              6519.53             0   \n",
       "1             439.112               240989              7640.43         58148   \n",
       "2             344.445               245301              6754.20        208276   \n",
       "3             308.562               193564              5542.66         81967   \n",
       "4             279.875               197625              5402.20        104813   \n",
       "7             234.859               191095              5240.15         71161   \n",
       "8             482.419               244314              6987.83         98483   \n",
       "10            314.058               224230              6554.51        254677   \n",
       "11            416.092               250204              6094.25             0   \n",
       "12            388.097               283465              7298.66         65282   \n",
       "\n",
       "    Num_transcripts  Num_proteins_hsapiens  Num_transcripts_hsapiens  \\\n",
       "0                 0                      0                         0   \n",
       "1           3619303                  47060                         0   \n",
       "2           8993166                 174117                   8993046   \n",
       "3           1989647                  45075                    344947   \n",
       "4           8780884                  51802                         0   \n",
       "7            758492                  62812                         0   \n",
       "8           3319583                  67352                         0   \n",
       "10           100392                 166881                         0   \n",
       "11                0                      0                         0   \n",
       "12                0                  65282                         0   \n",
       "\n",
       "    Num_rnaseq_reads  Num_tissues  Num_runs  \n",
       "0                  0            0         0  \n",
       "1        12069331098           42       433  \n",
       "2         3364726994           11        17  \n",
       "3         8324786874          198       264  \n",
       "4          670954419           10        45  \n",
       "7          403656856            1         1  \n",
       "8         8524329442           95       203  \n",
       "10        1697796066            4         9  \n",
       "11                 0            0         0  \n",
       "12        9134207560           37       107  \n",
       "\n",
       "[10 rows x 48 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Read Files of Genome Variables and Metadata\n",
    "metadata = pd.read_csv('Metadata.csv', header=0, sep='\\t', low_memory=False)\n",
    "metadata_assembly = pd.read_csv('Metadata_Assembly.csv', header=0, sep='\\t', low_memory=False)\n",
    "genome_variables = pd.read_csv('Output_Genome_Variables.csv', header=0, sep='\\t', low_memory=False)\n",
    "refseq_eukaryotes = pd.read_csv('Refseq_Eukaryotes.csv', header=0, sep='\\t', low_memory=False)\n",
    "\n",
    "#Merge all variables into a single File called 'Genome_Variables.csv'\n",
    "\n",
    "#Assembly accession name\n",
    "list_assembly_accession = []\n",
    "list_organism_name = []\n",
    "list_asm_name = []\n",
    "list_genome_size = []\n",
    "list_total_gene_count = []\n",
    "list_protein_coding_gene_count = []\n",
    "list_non_coding_gene_count = []\n",
    "list_Taxonomy = []\n",
    "list_genome_size_ungapped = []\n",
    "list_scaffold_count = []\n",
    "list_contig_count = []\n",
    "\n",
    "\n",
    "#We first add the variables of interst from \"Refseq_Eukaryotes.csv\" to \"genome_variables\"\n",
    "it=list(range(0, len(genome_variables)))   \n",
    "for i in it:\n",
    "    posf=len(genome_variables.FileName[i])\n",
    "    substring = genome_variables.FileName[i][4:posf]\n",
    "    pos=substring.find('_')+4\n",
    "    assembly_accession = genome_variables.FileName[i][0:pos]\n",
    "    list_assembly_accession.append(assembly_accession)\n",
    "    row=refseq_eukaryotes.loc[refseq_eukaryotes['assembly_accession'] == assembly_accession]\n",
    "    \n",
    "    organism_name=row.organism_name.tolist()[0]\n",
    "    list_organism_name.append(organism_name)\n",
    "    \n",
    "    asm_name=row.asm_name.tolist()[0]\n",
    "    list_asm_name.append(asm_name)\n",
    "    \n",
    "    genome_size=row.genome_size.tolist()[0]\n",
    "    list_genome_size.append(genome_size)\n",
    "    \n",
    "    total_gene_count=row.total_gene_count.tolist()[0]\n",
    "    list_total_gene_count.append(total_gene_count)\n",
    "    \n",
    "    protein_coding_gene_count=row.protein_coding_gene_count.tolist()[0]\n",
    "    list_protein_coding_gene_count.append(protein_coding_gene_count)\n",
    "    \n",
    "    non_coding_gene_count=row.non_coding_gene_count.tolist()[0]\n",
    "    list_non_coding_gene_count.append(non_coding_gene_count)\n",
    "    \n",
    "    Taxonomy=row.Taxonomy.tolist()[0]\n",
    "    list_Taxonomy.append(Taxonomy)\n",
    "    \n",
    "    genome_size_ungapped=row.genome_size_ungapped.tolist()[0]\n",
    "    list_genome_size_ungapped.append(genome_size_ungapped)\n",
    "    \n",
    "    scaffold_count=row.scaffold_count.tolist()[0]\n",
    "    list_scaffold_count.append(scaffold_count)\n",
    "    \n",
    "    contig_count=row.contig_count.tolist()[0]\n",
    "    list_contig_count.append(contig_count)\n",
    "       \n",
    "    \n",
    "\n",
    "        \n",
    "genome_variables[\"assembly_accession\"]=list_assembly_accession\n",
    "genome_variables[\"organism_name\"]=list_organism_name\n",
    "genome_variables[\"asm_name\"]=list_asm_name\n",
    "genome_variables[\"total_gene_count\"]=list_total_gene_count\n",
    "genome_variables[\"protein_coding_gene_count\"]=list_protein_coding_gene_count\n",
    "genome_variables[\"non_coding_gene_count\"]=list_non_coding_gene_count\n",
    "genome_variables[\"Taxonomy\"]=list_Taxonomy\n",
    "genome_variables[\"genome_size_ungapped\"]=list_genome_size_ungapped\n",
    "genome_variables[\"scaffold_count\"]=list_scaffold_count\n",
    "genome_variables[\"contig_count\"]=list_contig_count\n",
    "\n",
    "\n",
    "\n",
    "#We now add all data from \"Metadata Assembly\" to \"genome_variables\"---------------\n",
    "\n",
    "list_spanned_gaps = []\n",
    "list_unspanned_gaps = []\n",
    "list_scaffold_count_assembly = []\n",
    "list_scaffold_N50 = []\n",
    "list_contig_count_assembly = []\n",
    "list_contig_N50 = []\n",
    "list_total_gap_length = []\n",
    "list_gc_perc = []\n",
    "\n",
    "metadata_assembly = metadata_assembly.replace(r'\\n','', regex=True) \n",
    "it=list(range(0, len(genome_variables)))   \n",
    "for i in it:\n",
    "    accession=genome_variables.assembly_accession[i]\n",
    "    row=metadata_assembly.loc[metadata_assembly['refseq_assembly_accession'] == accession]\n",
    "    \n",
    "    spanned_gaps=row.spanned_gaps.tolist()[0]\n",
    "    list_spanned_gaps.append(spanned_gaps)\n",
    "    \n",
    "    unspanned_gaps=row.unspanned_gaps.tolist()[0]\n",
    "    list_unspanned_gaps.append(unspanned_gaps)\n",
    "    \n",
    "    scaffold_count=row.scaffold_count.tolist()[0]\n",
    "    list_scaffold_count_assembly.append(scaffold_count)\n",
    "    \n",
    "    scaffold_N50=row.scaffold_N50.tolist()[0]\n",
    "    list_scaffold_N50.append(scaffold_N50)\n",
    " \n",
    "    contig_count=row.contig_count.tolist()[0]\n",
    "    list_contig_count_assembly.append(contig_count)\n",
    "    \n",
    "    contig_N50=row.contig_N50.tolist()[0]\n",
    "    list_contig_N50.append(contig_N50)\n",
    "    \n",
    "    total_gap_length=row.total_gap_length.tolist()[0]\n",
    "    list_total_gap_length.append(total_gap_length)\n",
    "     \n",
    "    gc_perc=row.gc_perc.tolist()[0]\n",
    "    list_gc_perc.append(gc_perc)\n",
    "\n",
    "\n",
    "genome_variables[\"spanned_gaps\"]=list_spanned_gaps\n",
    "genome_variables[\"unspanned_gaps\"]=list_unspanned_gaps\n",
    "genome_variables[\"scaffold_count_assembly\"]=list_scaffold_count_assembly\n",
    "genome_variables[\"scaffold_N50\"]=list_scaffold_N50\n",
    "genome_variables[\"contig_count_assembly\"]=list_contig_count_assembly\n",
    "genome_variables[\"contig_N50\"]=list_contig_N50\n",
    "genome_variables[\"total_gap_length\"]=list_total_gap_length\n",
    "genome_variables[\"gc_perc\"]=list_gc_perc\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "#We finally add all data from \"Metadata\" to \"genome_variables\"---------------\n",
    "\n",
    "metadata = metadata.replace(r'\\n','', regex=True) \n",
    "\n",
    "\n",
    "list_Genes_total = []\n",
    "list_Genes_protein_coding = []\n",
    "list_mRNAs = []\n",
    "list_CDSs = []\n",
    "list_CDSs_model_refseq = []\n",
    "list_CDSs_known_refseq = []\n",
    "list_CDSs_fully_supported = []\n",
    "list_CDSs_ab_initio = []\n",
    "list_Length_count_genes = [] \n",
    "list_Length_mean_genes = []\n",
    "list_Length_count_transcripts = []\n",
    "list_Length_mean_transcripts = []\n",
    "list_Length_count_cds = []\n",
    "list_Length_mean_cds = []\n",
    "list_Length_count_exons = []\n",
    "list_Length_mean_exons = []\n",
    "list_Length_count_introns = []\n",
    "list_Length_mean_introns = []\n",
    "list_Num_proteins = []\n",
    "list_Num_transcripts = []\n",
    "list_Num_proteins_hsapiens = []\n",
    "list_Num_transcripts_hsapiens = []\n",
    "list_Num_rnaseq_reads = []\n",
    "list_Num_tissues = []\n",
    "list_Num_runs = []\n",
    "\n",
    "list_remove= []\n",
    "list_remove_accession = []\n",
    "\n",
    "it=list(range(0, len(genome_variables)))   \n",
    "for i in it:\n",
    "    accession=genome_variables.assembly_accession[i]\n",
    "    row=metadata.loc[metadata['Accession'] == accession]\n",
    "    \n",
    "    inv=1\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        list_remove_accession.append(i) \n",
    "        \n",
    "        accession=genome_variables.organism_name[i]\n",
    "        rowc=metadata.loc[metadata['ScientificName'] == accession]\n",
    "        if(len(rowc.ScientificName.tolist())==0):\n",
    "            list_remove.append(i)\n",
    "            inv=0            \n",
    "        else:\n",
    "            row=rowc\n",
    "\n",
    "            \n",
    "        \n",
    "        \n",
    "        \n",
    "          \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Genes_total = 0\n",
    "    else:\n",
    "        Genes_total=row.Genes_total.tolist()[0]\n",
    "    list_Genes_total.append(Genes_total)\n",
    "    \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Genes_protein_coding = 0\n",
    "    else:\n",
    "        Genes_protein_coding=row.Genes_protein_coding.tolist()[0]\n",
    "    list_Genes_protein_coding.append(Genes_protein_coding)\n",
    "    \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        mRNAs = 0\n",
    "    else:\n",
    "        mRNAs=row.mRNAs.tolist()[0]\n",
    "    list_mRNAs.append(mRNAs)\n",
    "    \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        CDSs = 0\n",
    "    else:\n",
    "        CDSs=row.CDSs.tolist()[0]\n",
    "    list_CDSs.append(CDSs)\n",
    "    \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        CDSs_model_refseq = 0\n",
    "    else:\n",
    "        CDSs_model_refseq=row.CDSs_model_refseq.tolist()[0]\n",
    "    list_CDSs_model_refseq.append(CDSs_model_refseq)\n",
    "    \n",
    " \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        CDSs_known_refseq = 0\n",
    "    else:\n",
    "        CDSs_known_refseq=row.CDSs_known_refseq.tolist()[0]\n",
    "    list_CDSs_known_refseq.append(CDSs_known_refseq)\n",
    "    \n",
    " \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        CDSs_fully_supported = 0\n",
    "    else:\n",
    "        CDSs_fully_supported=row.CDSs_fully_supported.tolist()[0]\n",
    "    list_CDSs_fully_supported.append(CDSs_fully_supported)\n",
    "    \n",
    " \n",
    "   \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        CDSs_ab_initio = 0\n",
    "    else:\n",
    "        CDSs_ab_initio=row.CDSs_ab_initio.tolist()[0]\n",
    "    list_CDSs_ab_initio.append(CDSs_ab_initio)\n",
    "    \n",
    "    \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_count_genes = 0\n",
    "    else:\n",
    "        Length_count_genes=row.Length_count_genes.tolist()[0]\n",
    "    list_Length_count_genes.append(Length_count_genes)\n",
    "    \n",
    " \n",
    "   \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_mean_genes = 0\n",
    "    else:\n",
    "        Length_mean_genes=row.Length_mean_genes.tolist()[0]\n",
    "    list_Length_mean_genes.append(Length_mean_genes)\n",
    "    \n",
    "    \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_count_transcripts = 0\n",
    "    else:\n",
    "        Length_count_transcripts=row.Length_count_transcripts.tolist()[0]\n",
    "    list_Length_count_transcripts.append(Length_count_transcripts)\n",
    "     \n",
    "   \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_mean_transcripts = 0\n",
    "    else:\n",
    "        Length_mean_transcripts=row.Length_mean_transcripts.tolist()[0]\n",
    "    list_Length_mean_transcripts.append(Length_mean_transcripts)\n",
    "    \n",
    " \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_count_cds = 0\n",
    "    else:\n",
    "        Length_count_cds=row.Length_count_cds.tolist()[0]\n",
    "    list_Length_count_cds.append(Length_count_cds)\n",
    "    \n",
    " \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_mean_cds = 0\n",
    "    else:\n",
    "        Length_mean_cds=row.Length_mean_cds.tolist()[0]\n",
    "    list_Length_mean_cds.append(Length_mean_cds)\n",
    "\n",
    "   \n",
    "    \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_count_exons = 0\n",
    "    else:\n",
    "        Length_count_exons=row.Length_count_exons.tolist()[0]\n",
    "    list_Length_count_exons.append(Length_count_exons)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_mean_exons = 0\n",
    "    else:\n",
    "        Length_mean_exons=row.Length_mean_exons.tolist()[0]\n",
    "    list_Length_mean_exons.append(Length_mean_exons)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_count_introns = 0\n",
    "    else:\n",
    "        Length_count_introns=row.Length_count_introns.tolist()[0]\n",
    "    list_Length_count_introns.append(Length_count_introns)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Length_mean_introns = 0\n",
    "    else:\n",
    "        Length_mean_introns=row.Length_mean_introns.tolist()[0]\n",
    "    list_Length_mean_introns.append(Length_mean_introns)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_proteins = 0\n",
    "    else:\n",
    "        Num_proteins=row.Num_proteins.tolist()[0]\n",
    "    list_Num_proteins.append(Num_proteins)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_transcripts = 0\n",
    "    else:\n",
    "        Num_transcripts=row.Num_transcripts.tolist()[0]\n",
    "    list_Num_transcripts.append(Num_transcripts)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_proteins_hsapiens = 0\n",
    "    else:\n",
    "        Num_proteins_hsapiens=row.Num_proteins_hsapiens.tolist()[0]\n",
    "    list_Num_proteins_hsapiens.append(Num_proteins_hsapiens)\n",
    "    \n",
    " \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_transcripts_hsapiens = 0\n",
    "    else:\n",
    "        Num_transcripts_hsapiens=row.Num_transcripts_hsapiens.tolist()[0]\n",
    "    list_Num_transcripts_hsapiens.append(Num_transcripts_hsapiens)\n",
    "    \n",
    "        \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_rnaseq_reads = 0\n",
    "    else:\n",
    "        Num_rnaseq_reads=row.Num_rnaseq_reads.tolist()[0]\n",
    "    list_Num_rnaseq_reads.append(Num_rnaseq_reads)\n",
    "    \n",
    " \n",
    "\n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_tissues = 0\n",
    "    else:\n",
    "        Num_tissues=row.Num_tissues.tolist()[0]\n",
    "    list_Num_tissues.append(Num_tissues)\n",
    "    \n",
    " \n",
    "    if(len(row.Accession.tolist())==0):\n",
    "        Num_runs = 0\n",
    "    else:\n",
    "        Num_runs=row.Num_runs.tolist()[0]\n",
    "    list_Num_runs.append(Num_runs)\n",
    "    \n",
    "         \n",
    "       \n",
    "\n",
    "genome_variables[\"Genes_total\"]=list_Genes_total\n",
    "genome_variables[\"Genes_protein_coding\"]=list_Genes_protein_coding\n",
    "genome_variables[\"mRNAs\"]=list_mRNAs\n",
    "genome_variables[\"CDSs\"]=list_CDSs\n",
    "genome_variables[\"CDSs_model_refseq\"]=list_CDSs_model_refseq\n",
    "genome_variables[\"CDSs_known_refseq\"]=list_CDSs_known_refseq\n",
    "genome_variables[\"CDSs_fully_supported\"]=list_CDSs_fully_supported\n",
    "genome_variables[\"CDSs_ab_initio\"]=list_CDSs_ab_initio\n",
    "genome_variables[\"Length_count_genes\"]=list_Length_count_genes\n",
    "genome_variables[\"Length_mean_genes\"]=list_Length_mean_genes\n",
    "genome_variables[\"Length_count_transcripts\"]=list_Length_count_transcripts\n",
    "genome_variables[\"Length_mean_transcripts\"]=list_Length_mean_transcripts\n",
    "genome_variables[\"Length_count_cds\"]=list_Length_count_cds\n",
    "genome_variables[\"Length_mean_cds\"]=list_Length_mean_cds\n",
    "genome_variables[\"Length_count_exons\"]=list_Length_count_exons\n",
    "genome_variables[\"Length_mean_exons\"]=list_Length_mean_exons\n",
    "genome_variables[\"Length_count_introns\"]=list_Length_count_introns\n",
    "genome_variables[\"Length_mean_introns\"]=list_Length_mean_introns\n",
    "genome_variables[\"Num_proteins\"]=list_Num_proteins\n",
    "genome_variables[\"Num_transcripts\"]=list_Num_transcripts\n",
    "genome_variables[\"Num_proteins_hsapiens\"]=list_Num_proteins_hsapiens\n",
    "genome_variables[\"Num_transcripts_hsapiens\"]=list_Num_transcripts_hsapiens\n",
    "genome_variables[\"Num_rnaseq_reads\"]=list_Num_rnaseq_reads\n",
    "genome_variables[\"Num_tissues\"]=list_Num_tissues\n",
    "genome_variables[\"Num_runs\"]=list_Num_runs\n",
    "\n",
    "\n",
    "genome_variables.drop(list_remove, inplace=True)\n",
    "genome_variables.to_csv('Genome_Variables.csv', sep='\\t')\n",
    "genome_variables[0:10]\n",
    "\n",
    "\n",
    "\n",
    "        \n",
    "                \n",
    "\n",
    "        \n",
    "        \n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "342e9ec4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ASR</th>\n",
       "      <th>num_gaps</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>scaffold_N50</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>contig_N50</th>\n",
       "      <th>gap_length</th>\n",
       "      <th>gap_length_per</th>\n",
       "      <th>CDSs</th>\n",
       "      <th>CDSs_model</th>\n",
       "      <th>...</th>\n",
       "      <th>Length_mean_cds</th>\n",
       "      <th>Length_mean_exons</th>\n",
       "      <th>Length_mean_introns</th>\n",
       "      <th>Num_proteins</th>\n",
       "      <th>Num_transcripts</th>\n",
       "      <th>Num_proteins_hsapiens</th>\n",
       "      <th>Num_transcripts_hsapiens</th>\n",
       "      <th>Num_rnaseq_reads</th>\n",
       "      <th>Num_tissues</th>\n",
       "      <th>Num_runs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5.327877</td>\n",
       "      <td>347</td>\n",
       "      <td>102</td>\n",
       "      <td>106145001</td>\n",
       "      <td>21254</td>\n",
       "      <td>59462871</td>\n",
       "      <td>73600614</td>\n",
       "      <td>0.026977</td>\n",
       "      <td>97829</td>\n",
       "      <td>48540</td>\n",
       "      <td>...</td>\n",
       "      <td>2051.09</td>\n",
       "      <td>482.807</td>\n",
       "      <td>6519.53</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3.703144</td>\n",
       "      <td>506</td>\n",
       "      <td>706</td>\n",
       "      <td>88231837</td>\n",
       "      <td>1307</td>\n",
       "      <td>48231277</td>\n",
       "      <td>29864641</td>\n",
       "      <td>0.011937</td>\n",
       "      <td>63699</td>\n",
       "      <td>59448</td>\n",
       "      <td>...</td>\n",
       "      <td>2084.54</td>\n",
       "      <td>439.112</td>\n",
       "      <td>7640.43</td>\n",
       "      <td>58148</td>\n",
       "      <td>3619303</td>\n",
       "      <td>47060</td>\n",
       "      <td>0</td>\n",
       "      <td>12069331098</td>\n",
       "      <td>42</td>\n",
       "      <td>433</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.540079</td>\n",
       "      <td>43306</td>\n",
       "      <td>7678</td>\n",
       "      <td>108171978</td>\n",
       "      <td>50982</td>\n",
       "      <td>210702</td>\n",
       "      <td>101087183</td>\n",
       "      <td>0.040640</td>\n",
       "      <td>59172</td>\n",
       "      <td>58994</td>\n",
       "      <td>...</td>\n",
       "      <td>2119.73</td>\n",
       "      <td>344.445</td>\n",
       "      <td>6754.20</td>\n",
       "      <td>208276</td>\n",
       "      <td>8993166</td>\n",
       "      <td>174117</td>\n",
       "      <td>8993046</td>\n",
       "      <td>3364726994</td>\n",
       "      <td>11</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.008855</td>\n",
       "      <td>253738</td>\n",
       "      <td>32</td>\n",
       "      <td>106310653</td>\n",
       "      <td>253769</td>\n",
       "      <td>28375</td>\n",
       "      <td>198136445</td>\n",
       "      <td>0.074098</td>\n",
       "      <td>36106</td>\n",
       "      <td>35979</td>\n",
       "      <td>...</td>\n",
       "      <td>1820.67</td>\n",
       "      <td>308.562</td>\n",
       "      <td>5542.66</td>\n",
       "      <td>81967</td>\n",
       "      <td>1989647</td>\n",
       "      <td>45075</td>\n",
       "      <td>344947</td>\n",
       "      <td>8324786874</td>\n",
       "      <td>198</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2.229472</td>\n",
       "      <td>180677</td>\n",
       "      <td>6450</td>\n",
       "      <td>17270019</td>\n",
       "      <td>187012</td>\n",
       "      <td>21250</td>\n",
       "      <td>183019268</td>\n",
       "      <td>0.080014</td>\n",
       "      <td>37971</td>\n",
       "      <td>37783</td>\n",
       "      <td>...</td>\n",
       "      <td>2002.99</td>\n",
       "      <td>279.875</td>\n",
       "      <td>5402.20</td>\n",
       "      <td>104813</td>\n",
       "      <td>8780884</td>\n",
       "      <td>51802</td>\n",
       "      <td>0</td>\n",
       "      <td>670954419</td>\n",
       "      <td>10</td>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1.805260</td>\n",
       "      <td>25787</td>\n",
       "      <td>8094</td>\n",
       "      <td>150810825</td>\n",
       "      <td>8094</td>\n",
       "      <td>181061</td>\n",
       "      <td>42905630</td>\n",
       "      <td>0.017734</td>\n",
       "      <td>34558</td>\n",
       "      <td>34516</td>\n",
       "      <td>...</td>\n",
       "      <td>1735.25</td>\n",
       "      <td>234.859</td>\n",
       "      <td>5240.15</td>\n",
       "      <td>71161</td>\n",
       "      <td>758492</td>\n",
       "      <td>62812</td>\n",
       "      <td>0</td>\n",
       "      <td>403656856</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3.620578</td>\n",
       "      <td>386</td>\n",
       "      <td>1958</td>\n",
       "      <td>103308737</td>\n",
       "      <td>1957</td>\n",
       "      <td>26402946</td>\n",
       "      <td>28162</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>65084</td>\n",
       "      <td>50772</td>\n",
       "      <td>...</td>\n",
       "      <td>2053.02</td>\n",
       "      <td>482.419</td>\n",
       "      <td>6987.83</td>\n",
       "      <td>98483</td>\n",
       "      <td>3319583</td>\n",
       "      <td>67352</td>\n",
       "      <td>0</td>\n",
       "      <td>8524329442</td>\n",
       "      <td>95</td>\n",
       "      <td>203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2.906335</td>\n",
       "      <td>121387</td>\n",
       "      <td>12169</td>\n",
       "      <td>122182240</td>\n",
       "      <td>133554</td>\n",
       "      <td>42877</td>\n",
       "      <td>104912766</td>\n",
       "      <td>0.042029</td>\n",
       "      <td>51779</td>\n",
       "      <td>51712</td>\n",
       "      <td>...</td>\n",
       "      <td>1911.76</td>\n",
       "      <td>314.058</td>\n",
       "      <td>6554.51</td>\n",
       "      <td>254677</td>\n",
       "      <td>100392</td>\n",
       "      <td>166881</td>\n",
       "      <td>0</td>\n",
       "      <td>1697796066</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3.716273</td>\n",
       "      <td>6286</td>\n",
       "      <td>4701</td>\n",
       "      <td>87230776</td>\n",
       "      <td>4700</td>\n",
       "      <td>1502753</td>\n",
       "      <td>9435464</td>\n",
       "      <td>0.003764</td>\n",
       "      <td>61026</td>\n",
       "      <td>58838</td>\n",
       "      <td>...</td>\n",
       "      <td>2229.31</td>\n",
       "      <td>416.092</td>\n",
       "      <td>6094.25</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>4.217887</td>\n",
       "      <td>69</td>\n",
       "      <td>159</td>\n",
       "      <td>64250934</td>\n",
       "      <td>159</td>\n",
       "      <td>40716615</td>\n",
       "      <td>33679</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>74854</td>\n",
       "      <td>74728</td>\n",
       "      <td>...</td>\n",
       "      <td>2034.31</td>\n",
       "      <td>388.097</td>\n",
       "      <td>7298.66</td>\n",
       "      <td>65282</td>\n",
       "      <td>0</td>\n",
       "      <td>65282</td>\n",
       "      <td>0</td>\n",
       "      <td>9134207560</td>\n",
       "      <td>37</td>\n",
       "      <td>107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2.454285</td>\n",
       "      <td>19800</td>\n",
       "      <td>15308</td>\n",
       "      <td>130230028</td>\n",
       "      <td>15307</td>\n",
       "      <td>310061</td>\n",
       "      <td>47192329</td>\n",
       "      <td>0.016332</td>\n",
       "      <td>52610</td>\n",
       "      <td>52434</td>\n",
       "      <td>...</td>\n",
       "      <td>1636.67</td>\n",
       "      <td>314.862</td>\n",
       "      <td>7121.27</td>\n",
       "      <td>215391</td>\n",
       "      <td>10539846</td>\n",
       "      <td>180592</td>\n",
       "      <td>9007415</td>\n",
       "      <td>88348496</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3.909689</td>\n",
       "      <td>244</td>\n",
       "      <td>2979</td>\n",
       "      <td>82346004</td>\n",
       "      <td>3267</td>\n",
       "      <td>46608966</td>\n",
       "      <td>34438797</td>\n",
       "      <td>0.011590</td>\n",
       "      <td>68172</td>\n",
       "      <td>62208</td>\n",
       "      <td>...</td>\n",
       "      <td>2093.71</td>\n",
       "      <td>374.472</td>\n",
       "      <td>7271.66</td>\n",
       "      <td>218406</td>\n",
       "      <td>12529703</td>\n",
       "      <td>182483</td>\n",
       "      <td>9014916</td>\n",
       "      <td>12841591966</td>\n",
       "      <td>45</td>\n",
       "      <td>442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3.244965</td>\n",
       "      <td>302</td>\n",
       "      <td>1250</td>\n",
       "      <td>104466507</td>\n",
       "      <td>1250</td>\n",
       "      <td>26764281</td>\n",
       "      <td>1636497</td>\n",
       "      <td>0.000610</td>\n",
       "      <td>54287</td>\n",
       "      <td>54075</td>\n",
       "      <td>...</td>\n",
       "      <td>2153.18</td>\n",
       "      <td>340.255</td>\n",
       "      <td>6674.18</td>\n",
       "      <td>52406</td>\n",
       "      <td>2065320</td>\n",
       "      <td>52350</td>\n",
       "      <td>0</td>\n",
       "      <td>9731777244</td>\n",
       "      <td>26</td>\n",
       "      <td>244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>2.522202</td>\n",
       "      <td>3367</td>\n",
       "      <td>647</td>\n",
       "      <td>274391693</td>\n",
       "      <td>647</td>\n",
       "      <td>2711962</td>\n",
       "      <td>2424778</td>\n",
       "      <td>0.001025</td>\n",
       "      <td>84642</td>\n",
       "      <td>83488</td>\n",
       "      <td>...</td>\n",
       "      <td>1908.80</td>\n",
       "      <td>290.983</td>\n",
       "      <td>4350.29</td>\n",
       "      <td>96683</td>\n",
       "      <td>9034822</td>\n",
       "      <td>57871</td>\n",
       "      <td>0</td>\n",
       "      <td>4834348345</td>\n",
       "      <td>49</td>\n",
       "      <td>170</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>2.983466</td>\n",
       "      <td>147442</td>\n",
       "      <td>8524</td>\n",
       "      <td>115033041</td>\n",
       "      <td>155965</td>\n",
       "      <td>30111</td>\n",
       "      <td>124546479</td>\n",
       "      <td>0.049572</td>\n",
       "      <td>54530</td>\n",
       "      <td>54219</td>\n",
       "      <td>...</td>\n",
       "      <td>2004.76</td>\n",
       "      <td>317.180</td>\n",
       "      <td>5951.02</td>\n",
       "      <td>118626</td>\n",
       "      <td>450288</td>\n",
       "      <td>62538</td>\n",
       "      <td>0</td>\n",
       "      <td>4257360031</td>\n",
       "      <td>37</td>\n",
       "      <td>117</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>2.496582</td>\n",
       "      <td>512</td>\n",
       "      <td>322</td>\n",
       "      <td>83338043</td>\n",
       "      <td>321</td>\n",
       "      <td>15146802</td>\n",
       "      <td>15158102</td>\n",
       "      <td>0.008153</td>\n",
       "      <td>38847</td>\n",
       "      <td>38469</td>\n",
       "      <td>...</td>\n",
       "      <td>2086.15</td>\n",
       "      <td>256.725</td>\n",
       "      <td>5280.63</td>\n",
       "      <td>108191</td>\n",
       "      <td>10835</td>\n",
       "      <td>61492</td>\n",
       "      <td>0</td>\n",
       "      <td>2342784260</td>\n",
       "      <td>20</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>2.952685</td>\n",
       "      <td>155</td>\n",
       "      <td>134</td>\n",
       "      <td>88025743</td>\n",
       "      <td>134</td>\n",
       "      <td>31900497</td>\n",
       "      <td>7546091</td>\n",
       "      <td>0.003635</td>\n",
       "      <td>48806</td>\n",
       "      <td>48673</td>\n",
       "      <td>...</td>\n",
       "      <td>2068.38</td>\n",
       "      <td>312.188</td>\n",
       "      <td>5696.37</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>2.819929</td>\n",
       "      <td>876</td>\n",
       "      <td>77</td>\n",
       "      <td>171742863</td>\n",
       "      <td>77</td>\n",
       "      <td>6892556</td>\n",
       "      <td>15088836</td>\n",
       "      <td>0.007155</td>\n",
       "      <td>47220</td>\n",
       "      <td>46999</td>\n",
       "      <td>...</td>\n",
       "      <td>2049.21</td>\n",
       "      <td>323.134</td>\n",
       "      <td>5401.68</td>\n",
       "      <td>95248</td>\n",
       "      <td>360230</td>\n",
       "      <td>59396</td>\n",
       "      <td>0</td>\n",
       "      <td>5983967570</td>\n",
       "      <td>20</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>2.636209</td>\n",
       "      <td>2291</td>\n",
       "      <td>1857</td>\n",
       "      <td>114273790</td>\n",
       "      <td>1856</td>\n",
       "      <td>3487364</td>\n",
       "      <td>1128377</td>\n",
       "      <td>0.000456</td>\n",
       "      <td>48592</td>\n",
       "      <td>48177</td>\n",
       "      <td>...</td>\n",
       "      <td>2017.17</td>\n",
       "      <td>311.261</td>\n",
       "      <td>5663.91</td>\n",
       "      <td>116719</td>\n",
       "      <td>125071</td>\n",
       "      <td>60843</td>\n",
       "      <td>79065</td>\n",
       "      <td>7778617470</td>\n",
       "      <td>61</td>\n",
       "      <td>185</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>2.417652</td>\n",
       "      <td>65245</td>\n",
       "      <td>63791</td>\n",
       "      <td>141453419</td>\n",
       "      <td>63791</td>\n",
       "      <td>75409</td>\n",
       "      <td>6638100</td>\n",
       "      <td>0.002820</td>\n",
       "      <td>43409</td>\n",
       "      <td>43269</td>\n",
       "      <td>...</td>\n",
       "      <td>1863.93</td>\n",
       "      <td>264.754</td>\n",
       "      <td>5505.92</td>\n",
       "      <td>76181</td>\n",
       "      <td>750211</td>\n",
       "      <td>53845</td>\n",
       "      <td>0</td>\n",
       "      <td>1508929024</td>\n",
       "      <td>14</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ASR  num_gaps  scaffold_count  scaffold_N50  contig_count  \\\n",
       "0   5.327877       347             102     106145001         21254   \n",
       "1   3.703144       506             706      88231837          1307   \n",
       "2   3.540079     43306            7678     108171978         50982   \n",
       "3   2.008855    253738              32     106310653        253769   \n",
       "4   2.229472    180677            6450      17270019        187012   \n",
       "5   1.805260     25787            8094     150810825          8094   \n",
       "6   3.620578       386            1958     103308737          1957   \n",
       "7   2.906335    121387           12169     122182240        133554   \n",
       "8   3.716273      6286            4701      87230776          4700   \n",
       "9   4.217887        69             159      64250934           159   \n",
       "10  2.454285     19800           15308     130230028         15307   \n",
       "11  3.909689       244            2979      82346004          3267   \n",
       "12  3.244965       302            1250     104466507          1250   \n",
       "13  2.522202      3367             647     274391693           647   \n",
       "14  2.983466    147442            8524     115033041        155965   \n",
       "15  2.496582       512             322      83338043           321   \n",
       "16  2.952685       155             134      88025743           134   \n",
       "17  2.819929       876              77     171742863            77   \n",
       "18  2.636209      2291            1857     114273790          1856   \n",
       "19  2.417652     65245           63791     141453419         63791   \n",
       "\n",
       "    contig_N50  gap_length  gap_length_per   CDSs  CDSs_model  ...  \\\n",
       "0     59462871    73600614        0.026977  97829       48540  ...   \n",
       "1     48231277    29864641        0.011937  63699       59448  ...   \n",
       "2       210702   101087183        0.040640  59172       58994  ...   \n",
       "3        28375   198136445        0.074098  36106       35979  ...   \n",
       "4        21250   183019268        0.080014  37971       37783  ...   \n",
       "5       181061    42905630        0.017734  34558       34516  ...   \n",
       "6     26402946       28162        0.000010  65084       50772  ...   \n",
       "7        42877   104912766        0.042029  51779       51712  ...   \n",
       "8      1502753     9435464        0.003764  61026       58838  ...   \n",
       "9     40716615       33679        0.000014  74854       74728  ...   \n",
       "10      310061    47192329        0.016332  52610       52434  ...   \n",
       "11    46608966    34438797        0.011590  68172       62208  ...   \n",
       "12    26764281     1636497        0.000610  54287       54075  ...   \n",
       "13     2711962     2424778        0.001025  84642       83488  ...   \n",
       "14       30111   124546479        0.049572  54530       54219  ...   \n",
       "15    15146802    15158102        0.008153  38847       38469  ...   \n",
       "16    31900497     7546091        0.003635  48806       48673  ...   \n",
       "17     6892556    15088836        0.007155  47220       46999  ...   \n",
       "18     3487364     1128377        0.000456  48592       48177  ...   \n",
       "19       75409     6638100        0.002820  43409       43269  ...   \n",
       "\n",
       "    Length_mean_cds  Length_mean_exons  Length_mean_introns  Num_proteins  \\\n",
       "0           2051.09            482.807              6519.53             0   \n",
       "1           2084.54            439.112              7640.43         58148   \n",
       "2           2119.73            344.445              6754.20        208276   \n",
       "3           1820.67            308.562              5542.66         81967   \n",
       "4           2002.99            279.875              5402.20        104813   \n",
       "5           1735.25            234.859              5240.15         71161   \n",
       "6           2053.02            482.419              6987.83         98483   \n",
       "7           1911.76            314.058              6554.51        254677   \n",
       "8           2229.31            416.092              6094.25             0   \n",
       "9           2034.31            388.097              7298.66         65282   \n",
       "10          1636.67            314.862              7121.27        215391   \n",
       "11          2093.71            374.472              7271.66        218406   \n",
       "12          2153.18            340.255              6674.18         52406   \n",
       "13          1908.80            290.983              4350.29         96683   \n",
       "14          2004.76            317.180              5951.02        118626   \n",
       "15          2086.15            256.725              5280.63        108191   \n",
       "16          2068.38            312.188              5696.37             0   \n",
       "17          2049.21            323.134              5401.68         95248   \n",
       "18          2017.17            311.261              5663.91        116719   \n",
       "19          1863.93            264.754              5505.92         76181   \n",
       "\n",
       "    Num_transcripts  Num_proteins_hsapiens  Num_transcripts_hsapiens  \\\n",
       "0                 0                      0                         0   \n",
       "1           3619303                  47060                         0   \n",
       "2           8993166                 174117                   8993046   \n",
       "3           1989647                  45075                    344947   \n",
       "4           8780884                  51802                         0   \n",
       "5            758492                  62812                         0   \n",
       "6           3319583                  67352                         0   \n",
       "7            100392                 166881                         0   \n",
       "8                 0                      0                         0   \n",
       "9                 0                  65282                         0   \n",
       "10         10539846                 180592                   9007415   \n",
       "11         12529703                 182483                   9014916   \n",
       "12          2065320                  52350                         0   \n",
       "13          9034822                  57871                         0   \n",
       "14           450288                  62538                         0   \n",
       "15            10835                  61492                         0   \n",
       "16                0                      0                         0   \n",
       "17           360230                  59396                         0   \n",
       "18           125071                  60843                     79065   \n",
       "19           750211                  53845                         0   \n",
       "\n",
       "    Num_rnaseq_reads  Num_tissues  Num_runs  \n",
       "0                  0            0         0  \n",
       "1        12069331098           42       433  \n",
       "2         3364726994           11        17  \n",
       "3         8324786874          198       264  \n",
       "4          670954419           10        45  \n",
       "5          403656856            1         1  \n",
       "6         8524329442           95       203  \n",
       "7         1697796066            4         9  \n",
       "8                  0            0         0  \n",
       "9         9134207560           37       107  \n",
       "10          88348496            1         2  \n",
       "11       12841591966           45       442  \n",
       "12        9731777244           26       244  \n",
       "13        4834348345           49       170  \n",
       "14        4257360031           37       117  \n",
       "15        2342784260           20        88  \n",
       "16                 0            0         0  \n",
       "17        5983967570           20        91  \n",
       "18        7778617470           61       185  \n",
       "19        1508929024           14        20  \n",
       "\n",
       "[20 rows x 29 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Now, from \"Genome_Variables.csv\", we select a new file with the variables to perform the Metadata Analisis\n",
    "\n",
    "genome_variables = pd.read_csv('Genome_Variables.csv', header=0, sep='\\t', low_memory=False)\n",
    "\n",
    "analisis_metadata_gaps = genome_variables.spanned_gaps+genome_variables.unspanned_gaps\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "analisis_metadata = pd.DataFrame()\n",
    "\n",
    "analisis_metadata[\"ASR\"]=genome_variables.ASR\n",
    "analisis_metadata[\"num_gaps\"]=analisis_metadata_gaps\n",
    "analisis_metadata[\"scaffold_count\"]=genome_variables.scaffold_count_assembly\n",
    "analisis_metadata[\"scaffold_N50\"]=genome_variables.scaffold_N50\n",
    "analisis_metadata[\"contig_count\"]=genome_variables.contig_count\n",
    "analisis_metadata[\"contig_N50\"]=genome_variables.contig_N50\n",
    "analisis_metadata[\"gap_length\"]=genome_variables.total_gap_length\n",
    "analisis_metadata[\"gap_length_per\"]=genome_variables.total_gap_length/genome_variables.GenomeSize\n",
    "analisis_metadata[\"CDSs\"]=genome_variables.CDSs\n",
    "analisis_metadata[\"CDSs_model\"]=genome_variables.CDSs_model_refseq\n",
    "analisis_metadata[\"CDSs_known\"]=genome_variables.CDSs_known_refseq\n",
    "analisis_metadata[\"CDSs_fully_supported\"]=genome_variables.CDSs_fully_supported\n",
    "analisis_metadata[\"CDSs_ab_initio\"]=genome_variables.CDSs_ab_initio\n",
    "analisis_metadata[\"CDSs_model_per\"]=genome_variables.CDSs_model_refseq/genome_variables.CDSs\n",
    "analisis_metadata[\"CDSs_known_per\"]=genome_variables.CDSs_known_refseq/genome_variables.CDSs\n",
    "analisis_metadata[\"CDSs_fully_supported_per\"]=genome_variables.CDSs_fully_supported/genome_variables.CDSs\n",
    "analisis_metadata[\"CDSs_ab_initio_per\"]=genome_variables.CDSs_ab_initio/genome_variables.CDSs\n",
    "analisis_metadata[\"Length_mean_genes\"]=genome_variables.Length_mean_genes\n",
    "analisis_metadata[\"Length_mean_transcripts\"]=genome_variables.Length_mean_transcripts\n",
    "analisis_metadata[\"Length_mean_cds\"]=genome_variables.Length_mean_cds\n",
    "analisis_metadata[\"Length_mean_exons\"]=genome_variables.Length_mean_exons\n",
    "analisis_metadata[\"Length_mean_introns\"]=genome_variables.Length_mean_introns\n",
    "analisis_metadata[\"Num_proteins\"]=genome_variables.Num_proteins\n",
    "analisis_metadata[\"Num_transcripts\"]=genome_variables.Num_transcripts\n",
    "analisis_metadata[\"Num_proteins_hsapiens\"]=genome_variables.Num_proteins_hsapiens\n",
    "analisis_metadata[\"Num_transcripts_hsapiens\"]=genome_variables.Num_transcripts_hsapiens\n",
    "analisis_metadata[\"Num_rnaseq_reads\"]=genome_variables.Num_rnaseq_reads\n",
    "analisis_metadata[\"Num_tissues\"]=genome_variables.Num_tissues\n",
    "analisis_metadata[\"Num_runs\"]=genome_variables.Num_runs\n",
    "\n",
    "\n",
    "analisis_metadata.to_csv('Analisis_Metadata.csv', sep='\\t')\n",
    "\n",
    "\n",
    "analisis_metadata[0:20]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8279c29a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f62cd628",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66896065",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
