{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "1d944d8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filename_assembly</th>\n",
       "      <th>assembly_name</th>\n",
       "      <th>refseq_assembly_accession</th>\n",
       "      <th>spanned_gaps</th>\n",
       "      <th>unspanned_gaps</th>\n",
       "      <th>scaffold_count</th>\n",
       "      <th>scaffold_N50</th>\n",
       "      <th>contig_count</th>\n",
       "      <th>contig_N50</th>\n",
       "      <th>total_gap_length</th>\n",
       "      <th>gc_perc</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_000001635.27_GRCm39_assembly_stats.txt</td>\n",
       "      <td>GRCm39\\n</td>\n",
       "      <td>GCF_000001635.27\\n</td>\n",
       "      <td>204\\n</td>\n",
       "      <td>143\\n</td>\n",
       "      <td>102\\n</td>\n",
       "      <td>106145001\\n</td>\n",
       "      <td>306\\n</td>\n",
       "      <td>59462871\\n</td>\n",
       "      <td>73600614\\n</td>\n",
       "      <td>42\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_000002035.6_GRCz11_assembly_stats.txt</td>\n",
       "      <td>GRCz11\\n</td>\n",
       "      <td>GCF_000002035.6\\n</td>\n",
       "      <td>17835\\n</td>\n",
       "      <td>925\\n</td>\n",
       "      <td>2848\\n</td>\n",
       "      <td>4737936\\n</td>\n",
       "      <td>20680\\n</td>\n",
       "      <td>854399\\n</td>\n",
       "      <td>4691662\\n</td>\n",
       "      <td>36.5\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_000002425.4_Phypa_V3_assembly_stats.txt</td>\n",
       "      <td>PhypaV3\\n</td>\n",
       "      <td>GCF_000002425.4\\n</td>\n",
       "      <td>2643\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>359\\n</td>\n",
       "      <td>17435539\\n</td>\n",
       "      <td>3002\\n</td>\n",
       "      <td>465069\\n</td>\n",
       "      <td>5980520\\n</td>\n",
       "      <td>33.5\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_000002775.5_P.trichocarpa_v4.1_assembly_st...</td>\n",
       "      <td>P.trichocarpa_v4.1\\n</td>\n",
       "      <td>GCF_000002775.5\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>59\\n</td>\n",
       "      <td>106\\n</td>\n",
       "      <td>13159747\\n</td>\n",
       "      <td>106\\n</td>\n",
       "      <td>13159747\\n</td>\n",
       "      <td>590000\\n</td>\n",
       "      <td>34\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_000003025.6_Sscrofa11.1_assembly_stats.txt</td>\n",
       "      <td>Sscrofa11.1\\n</td>\n",
       "      <td>GCF_000003025.6\\n</td>\n",
       "      <td>413\\n</td>\n",
       "      <td>93\\n</td>\n",
       "      <td>706\\n</td>\n",
       "      <td>88231837\\n</td>\n",
       "      <td>1118\\n</td>\n",
       "      <td>48231277\\n</td>\n",
       "      <td>29864641\\n</td>\n",
       "      <td>42\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>GCF_963924715.1_fThuThy2.1_assembly_stats.txt</td>\n",
       "      <td>fThuThy2.1\\n</td>\n",
       "      <td>GCF_963924715.1\\n</td>\n",
       "      <td>468\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>119\\n</td>\n",
       "      <td>34318521\\n</td>\n",
       "      <td>587\\n</td>\n",
       "      <td>2956553\\n</td>\n",
       "      <td>93600\\n</td>\n",
       "      <td>40\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>GCF_963930625.1_mMunRee1.1_assembly_stats.txt</td>\n",
       "      <td>mMunRee1.1\\n</td>\n",
       "      <td>GCF_963930625.1\\n</td>\n",
       "      <td>682\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>267\\n</td>\n",
       "      <td>114544513\\n</td>\n",
       "      <td>949\\n</td>\n",
       "      <td>5780196\\n</td>\n",
       "      <td>136400\\n</td>\n",
       "      <td>42\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>GCF_963930695.1_fLabBer1.1_assembly_stats.txt</td>\n",
       "      <td>fLabBer1.1\\n</td>\n",
       "      <td>GCF_963930695.1\\n</td>\n",
       "      <td>815\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>131\\n</td>\n",
       "      <td>31261685\\n</td>\n",
       "      <td>946\\n</td>\n",
       "      <td>2715806\\n</td>\n",
       "      <td>163000\\n</td>\n",
       "      <td>40.5\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>GCF_963932015.1_bAnaAcu1.1_assembly_stats.txt</td>\n",
       "      <td>bAnaAcu1.1\\n</td>\n",
       "      <td>GCF_963932015.1\\n</td>\n",
       "      <td>631\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>394\\n</td>\n",
       "      <td>76435830\\n</td>\n",
       "      <td>1025\\n</td>\n",
       "      <td>3416933\\n</td>\n",
       "      <td>125881\\n</td>\n",
       "      <td>42.5\\n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>GCF_963966145.1_icTenMoli1.1_assembly_stats.txt</td>\n",
       "      <td>icTenMoli1.1\\n</td>\n",
       "      <td>GCF_963966145.1\\n</td>\n",
       "      <td>71\\n</td>\n",
       "      <td>0\\n</td>\n",
       "      <td>236\\n</td>\n",
       "      <td>29772305\\n</td>\n",
       "      <td>307\\n</td>\n",
       "      <td>7033857\\n</td>\n",
       "      <td>14200\\n</td>\n",
       "      <td>36.5\\n</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>693 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     filename_assembly         assembly_name  \\\n",
       "0           GCF_000001635.27_GRCm39_assembly_stats.txt              GRCm39\\n   \n",
       "1            GCF_000002035.6_GRCz11_assembly_stats.txt              GRCz11\\n   \n",
       "2          GCF_000002425.4_Phypa_V3_assembly_stats.txt             PhypaV3\\n   \n",
       "3    GCF_000002775.5_P.trichocarpa_v4.1_assembly_st...  P.trichocarpa_v4.1\\n   \n",
       "4       GCF_000003025.6_Sscrofa11.1_assembly_stats.txt         Sscrofa11.1\\n   \n",
       "..                                                 ...                   ...   \n",
       "688      GCF_963924715.1_fThuThy2.1_assembly_stats.txt          fThuThy2.1\\n   \n",
       "689      GCF_963930625.1_mMunRee1.1_assembly_stats.txt          mMunRee1.1\\n   \n",
       "690      GCF_963930695.1_fLabBer1.1_assembly_stats.txt          fLabBer1.1\\n   \n",
       "691      GCF_963932015.1_bAnaAcu1.1_assembly_stats.txt          bAnaAcu1.1\\n   \n",
       "692    GCF_963966145.1_icTenMoli1.1_assembly_stats.txt        icTenMoli1.1\\n   \n",
       "\n",
       "    refseq_assembly_accession spanned_gaps unspanned_gaps scaffold_count  \\\n",
       "0          GCF_000001635.27\\n        204\\n          143\\n          102\\n   \n",
       "1           GCF_000002035.6\\n      17835\\n          925\\n         2848\\n   \n",
       "2           GCF_000002425.4\\n       2643\\n            0\\n          359\\n   \n",
       "3           GCF_000002775.5\\n          0\\n           59\\n          106\\n   \n",
       "4           GCF_000003025.6\\n        413\\n           93\\n          706\\n   \n",
       "..                        ...          ...            ...            ...   \n",
       "688         GCF_963924715.1\\n        468\\n            0\\n          119\\n   \n",
       "689         GCF_963930625.1\\n        682\\n            0\\n          267\\n   \n",
       "690         GCF_963930695.1\\n        815\\n            0\\n          131\\n   \n",
       "691         GCF_963932015.1\\n        631\\n            0\\n          394\\n   \n",
       "692         GCF_963966145.1\\n         71\\n            0\\n          236\\n   \n",
       "\n",
       "    scaffold_N50 contig_count  contig_N50 total_gap_length gc_perc  \n",
       "0    106145001\\n        306\\n  59462871\\n       73600614\\n    42\\n  \n",
       "1      4737936\\n      20680\\n    854399\\n        4691662\\n  36.5\\n  \n",
       "2     17435539\\n       3002\\n    465069\\n        5980520\\n  33.5\\n  \n",
       "3     13159747\\n        106\\n  13159747\\n         590000\\n    34\\n  \n",
       "4     88231837\\n       1118\\n  48231277\\n       29864641\\n    42\\n  \n",
       "..           ...          ...         ...              ...     ...  \n",
       "688   34318521\\n        587\\n   2956553\\n          93600\\n    40\\n  \n",
       "689  114544513\\n        949\\n   5780196\\n         136400\\n    42\\n  \n",
       "690   31261685\\n        946\\n   2715806\\n         163000\\n  40.5\\n  \n",
       "691   76435830\\n       1025\\n   3416933\\n         125881\\n  42.5\\n  \n",
       "692   29772305\\n        307\\n   7033857\\n          14200\\n  36.5\\n  \n",
       "\n",
       "[693 rows x 11 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import networkx as nx\n",
    "import csv\n",
    "from xml.dom import minidom\n",
    "import xml.etree.ElementTree as ET\n",
    "from xml.etree.ElementTree import tostring\n",
    "\n",
    "\n",
    "\n",
    "#-------- Variables --------------------------------------------------------------------\n",
    "\n",
    "list_filename = []\n",
    "list_name = []\n",
    "list_accession = []\n",
    "list_scientific_name = []\n",
    "list_genes_total = []\n",
    "list_genes_protein_coding = []\n",
    "list_mrna = []\n",
    "list_cds = []\n",
    "list_model_refseq = []\n",
    "list_known_refseq = []\n",
    "list_cds_fully_supported = []\n",
    "list_cds_ab_initio = []\n",
    "list_length_count_genes = []\n",
    "list_length_mean_genes = []\n",
    "list_length_count_transcripts = []\n",
    "list_length_mean_transcripts = []\n",
    "list_length_count_cds = []\n",
    "list_length_mean_cds = []\n",
    "list_length_count_introns = []\n",
    "list_length_mean_introns = []\n",
    "list_length_count_exons = []\n",
    "list_length_mean_exons = []\n",
    "list_proteins_seq = []\n",
    "list_transcripts_seq = []\n",
    "list_proteins_hsapiens_seq = []\n",
    "list_transcripts_hsapiens_seq = []\n",
    "list_rnaseq_reads = []\n",
    "list_num_tissues = []\n",
    "list_num_runs = []\n",
    "\n",
    "\n",
    "#-------- Read xml files ---------------------------------------------------------------\n",
    "list_filenames = pd.read_csv('File_names.csv', header=0, sep='\\t', low_memory=False)\n",
    "it=list(range(0, len(list_filenames.xml)))\n",
    "\n",
    "\n",
    "    \n",
    "for i in it:\n",
    "    #1\n",
    "    filename=list_filenames.xml.iloc[i]\n",
    "    list_filename.append(filename)\n",
    "\n",
    "    tree= ET.parse(filename)\n",
    "    root = tree.getroot()\n",
    "    \n",
    "    #2\n",
    "    m=root.findall(\"./AssembliesReport/FullAssembly/Name\")\n",
    "    name=m[0].text\n",
    "    list_name.append(name)\n",
    "\n",
    "    #3\n",
    "    m=root.findall(\"./AssembliesReport/FullAssembly/Accession\")\n",
    "    accession=m[0].text\n",
    "    list_accession.append(accession)\n",
    "\n",
    "    #4\n",
    "    m=root.findall(\".//ScientificName\")\n",
    "    if not m:\n",
    "        m='Error'\n",
    "    else:\n",
    "        m=m[0].text\n",
    "    scientific_name=m\n",
    "    list_scientific_name.append(scientific_name)\n",
    "    \n",
    "    #5\n",
    "    m = root.find(\".//Counts[@feature_type='genes']/Total\")\n",
    "    genes_total=m.text\n",
    "    list_genes_total.append(genes_total)\n",
    "    \n",
    "    #6\n",
    "    m = root.find(\".//Counts[@feature_type='genes']/Subcount[@subtype='protein coding']\")\n",
    "    genes_protein_coding=m.text\n",
    "    list_genes_protein_coding.append(genes_protein_coding)\n",
    "\n",
    "    #7\n",
    "    m = root.find(\".//Counts[@feature_type='mRNAs']/Total\")\n",
    "    mrna=m.text\n",
    "    list_mrna.append(mrna)\n",
    "    \n",
    "    #8\n",
    "    m = root.find(\".//Counts[@feature_type='CDSs']/Total\")\n",
    "    cds=m.text\n",
    "    list_cds.append(cds)\n",
    "    \n",
    "    #9\n",
    "    m = root.find(\".//Counts[@feature_type='CDSs']/Subcount[@subtype='model RefSeq']\")\n",
    "    model_refseq=m.text\n",
    "    list_model_refseq.append(model_refseq)\n",
    "    \n",
    "    \n",
    "    #10\n",
    "    m = root.find(\".//Counts[@feature_type='CDSs']/Subcount[@subtype='known RefSeq']\")\n",
    "    known_refseq=m.text\n",
    "    list_known_refseq.append(known_refseq)\n",
    "    \n",
    "    #11\n",
    "    m = root.find(\".//Counts[@feature_type='CDSs']/Subcount[@subtype='fully supported']\")\n",
    "    cds_fully_supported=m.text\n",
    "    list_cds_fully_supported.append(cds_fully_supported)\n",
    "    \n",
    "    \n",
    "    #12\n",
    "    m = root.find(\".//Counts[@feature_type='CDSs']/Subcount[@subtype='ab initio > 5%']\")\n",
    "    cds_ab_initio=m.text\n",
    "    list_cds_ab_initio.append(cds_ab_initio)\n",
    "    \n",
    "    #13\n",
    "    m = root.find(\".//LengthStats[@category='Genes']/Count\")\n",
    "    length_count_genes=m.text\n",
    "    list_length_count_genes.append(length_count_genes)\n",
    "    \n",
    "    #14\n",
    "    m = root.find(\".//LengthStats[@category='Genes']/Mean\")\n",
    "    length_mean_genes=m.text\n",
    "    list_length_mean_genes.append(length_mean_genes)\n",
    "    \n",
    "    #15\n",
    "    m = root.find(\".//LengthStats[@category='All transcripts']/Count\")\n",
    "    list_length_count_transcripts.append(m.text)\n",
    "        \n",
    "    #16\n",
    "    m = root.find(\".//LengthStats[@category='All transcripts']/Mean\")\n",
    "    list_length_mean_transcripts.append(m.text)\n",
    "    \n",
    "    #17\n",
    "    m = root.find(\".//LengthStats[@category='CDSs']/Count\")\n",
    "    list_length_count_cds.append(m.text)\n",
    "    \n",
    "    #18\n",
    "    m = root.find(\".//LengthStats[@category='CDSs']/Mean\")\n",
    "    list_length_mean_cds.append(m.text)\n",
    "    \n",
    "    #19\n",
    "    m = root.find(\".//LengthStats[@category='All exons']/Count\")\n",
    "    list_length_count_exons.append(m.text)\n",
    "    \n",
    "    #20\n",
    "    m = root.find(\".//LengthStats[@category='All exons']/Mean\")\n",
    "    list_length_mean_exons.append(m.text)\n",
    "    \n",
    "    #21\n",
    "    m = root.find(\".//LengthStats[@category='All introns']/Count\")\n",
    "    list_length_count_introns.append(m.text)\n",
    "    \n",
    "    #22\n",
    "    m = root.find(\".//LengthStats[@category='All introns']/Mean\")\n",
    "    list_length_mean_introns.append(m.text)\n",
    "\n",
    "    \n",
    "    #Find name Accession of Primary Assembly\n",
    "    mx=root.findall(\".//AssemblyUnit/Name\")  \n",
    "    n=\"Primary Assembly\"\n",
    "    it=list(range(0, len(mx)))\n",
    "    for i in it:\n",
    "        if mx[i].text==n:\n",
    "            #print(mx[i].text)\n",
    "            nm=root.findall(\".//AssemblyUnit/Accession\") \n",
    "            name=nm[i].text\n",
    "\n",
    "            \n",
    "    #Number of proteins from Primary Assembly\n",
    "    #mx = root.find(\".//ProteinAlignStats\")  \n",
    "    mx=root.find(f'.//ProteinAlignStats/AssemblyStats[@accession=\"{name}\"]')  \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//TotalSequences\")\n",
    "        it=list(range(0, len(m)))\n",
    "        x=[m[i].text for i in it]\n",
    "        m=[int(k) for k in x]\n",
    "        num_seq=sum(m)\n",
    "    else:\n",
    "        #print(\"Variable is None\")\n",
    "        num_seq=0\n",
    "    list_proteins_seq.append(num_seq)\n",
    "    \n",
    "    \n",
    "    #Number of proteins from Primary Assembly with \"Species = Homo Sapiens\"\n",
    "    mx=root.find(f'.//ProteinAlignStats/AssemblyStats[@accession=\"{name}\"]')  \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//CategoryStats[@Species='Homo sapiens']/TotalSequences\")\n",
    "        #print(m[0].text)\n",
    "        it=list(range(0, len(m)))\n",
    "        x=[m[i].text for i in it]\n",
    "        m=[int(k) for k in x]\n",
    "        num_seq=sum(m)\n",
    "    else:\n",
    "        #print(\"Variable is None\")\n",
    "        num_seq=0\n",
    "    list_proteins_hsapiens_seq.append(num_seq)\n",
    "\n",
    "    \n",
    "    \n",
    "    #Number of transcripts from Primary Assembly\n",
    "    mx=root.find(f'.//TranscriptAlignStats/AssemblyStats[@accession=\"{name}\"]')     \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m=mx.findall(\".//TotalSequences\")\n",
    "        it=list(range(0, len(m)))\n",
    "        x=[m[i].text for i in it]\n",
    "        m=[int(k) for k in x]\n",
    "        num_seq=sum(m)\n",
    "    else:\n",
    "        #print(\"Variable is None\")\n",
    "        num_seq=0\n",
    "    list_transcripts_seq.append(num_seq)\n",
    "    \n",
    "\n",
    "    #Number of transcripts from Primary Assembly with \"Species = Homo Sapiens\"\n",
    "    mx=root.find(f'.//TranscriptAlignStats/AssemblyStats[@accession=\"{name}\"]')     \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//CategoryStats[@Species='Homo sapiens']/TotalSequences\")\n",
    "        #print(m[0].text)\n",
    "        it=list(range(0, len(m)))\n",
    "        x=[m[i].text for i in it]\n",
    "        m=[int(k) for k in x]\n",
    "        num_seq=sum(m)\n",
    "    else:\n",
    "        #print(\"Variable is None\")\n",
    "        num_seq=0\n",
    "    list_transcripts_hsapiens_seq.append(num_seq)\n",
    "    \n",
    "    \n",
    "    #Num RNA-Seq reads\n",
    "    mx=root.find(f'.//RnaseqAlignReport/AssemblyUnit[@accession=\"{name}\"]')  \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//SampleStats[@sample='multiple']/NumReads\")\n",
    "        num_seq=m[0].text\n",
    "    else:\n",
    "        num_seq=0\n",
    "    list_rnaseq_reads.append(num_seq)\n",
    "    \n",
    "    \n",
    "    #Number of tissues\n",
    "    list_tissues = []\n",
    "    mx=root.find(f'.//RnaseqAlignReport/AssemblyUnit[@accession=\"{name}\"]')  \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//SampleName\")\n",
    "        itx=list(range(0, len(m)))\n",
    "        for j in itx:\n",
    "            tissue=m[j].text\n",
    "            pos=tissue.find('(')\n",
    "            start=0\n",
    "            substring = tissue[start:pos]\n",
    "            list_tissues.append(substring)\n",
    "        list_tissues = list(set(list_tissues)) \n",
    "        list_tissues.remove(\"Aggregate of all aligned sample\")\n",
    "        #print(list_tissues)\n",
    "        num_tissues=len(list_tissues)\n",
    "    else:\n",
    "        num_tissues=0      \n",
    "    list_num_tissues.append(num_tissues)\n",
    "    \n",
    "    \n",
    "    #Number of runs\n",
    "    mx=root.find(f'.//RnaseqAlignReport/AssemblyUnit[@accession=\"{name}\"]')  \n",
    "    if mx is not None:\n",
    "        #print(\"Variable is not None\")\n",
    "        m = mx.findall(\".//RunStats\")\n",
    "        num_runs=len(m)\n",
    "    else:\n",
    "        num_runs=0 \n",
    "    list_num_runs.append(num_runs)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "            \n",
    "#-------- Output Data ---------------------------------------------------------------\n",
    "data = pd.DataFrame()\n",
    "data = pd.DataFrame(data,columns=['FileName', 'Name','Accession','ScientificName','Genes_total','Genes_protein_coding','mRNAs','CDSs','CDSs_model_refseq','CDSs_known_refseq','CDSs_fully_supported','CDSs_ab_initio','Length_count_genes','Length_mean_genes','Length_count_transcripts','Length_mean_transcripts','Length_count_cds','Length_mean_cds','Length_count_exons','Length_mean_exons','Length_count_introns','Length_mean_introns',\n",
    "'Num_proteins','Num_transcripts','Num_proteins_hsapiens','Num_transcripts_hsapiens','Num_rnaseq_reads','Num_tissues','Num_runs'])\n",
    "\n",
    "\n",
    "    \n",
    "data.FileName=list_filename\n",
    "data.Name=list_name\n",
    "data.Accession=list_accession\n",
    "data.ScientificName=list_scientific_name\n",
    "data.Genes_total=list_genes_total\n",
    "data.Genes_protein_coding=list_genes_protein_coding\n",
    "data.mRNAs=list_mrna\n",
    "data.CDSs=list_cds\n",
    "data.CDSs_model_refseq=list_model_refseq\n",
    "data.CDSs_known_refseq=list_known_refseq\n",
    "data.CDSs_fully_supported=list_cds_fully_supported\n",
    "data.CDSs_ab_initio=list_cds_ab_initio\n",
    "data.Length_count_genes=list_length_count_genes\n",
    "data.Length_mean_genes=list_length_mean_genes\n",
    "data.Length_count_transcripts=list_length_count_transcripts\n",
    "data.Length_mean_transcripts=list_length_mean_transcripts\n",
    "data.Length_count_cds=list_length_count_cds\n",
    "data.Length_mean_cds=list_length_mean_cds\n",
    "data.Length_count_introns=list_length_count_introns\n",
    "data.Length_mean_introns=list_length_mean_introns\n",
    "data.Length_count_exons=list_length_count_exons\n",
    "data.Length_mean_exons=list_length_mean_exons  \n",
    "data.Num_proteins_hsapiens=list_proteins_hsapiens_seq\n",
    "data.Num_transcripts_hsapiens=list_transcripts_hsapiens_seq   \n",
    "data.Num_proteins=list_proteins_seq\n",
    "data.Num_transcripts=list_transcripts_seq\n",
    "data.Num_rnaseq_reads=list_rnaseq_reads\n",
    "data.Num_tissues=list_num_tissues\n",
    "data.Num_runs=list_num_runs\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "    \n",
    "data.to_csv('Metadata.csv', sep='\\t')\n",
    "#data\n",
    "    \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "#-------- Read Assembly txt files --------------------------------------------------\n",
    "\n",
    "list_filename_assembly = []\n",
    "list_assembly_name = []\n",
    "list_refseq_assembly_accession = []\n",
    "list_spanned_gaps = []\n",
    "list_unspanned_gaps = []\n",
    "list_scaffold_count = []\n",
    "list_scaffold_N50 = []\n",
    "list_contig_count = []\n",
    "list_contig_N50 = []\n",
    "list_total_gap_length = []\n",
    "list_gc_perc = []\n",
    "            \n",
    "  \n",
    "list_filenames = pd.read_csv('/home/rebeca/Escritorio/Data/Assembly/File_names.csv', header=0, sep='\\t', low_memory=False)\n",
    "it=list(range(0, len(list_filenames)))\n",
    "\n",
    "for i in it:\n",
    "    \n",
    "    filename=list_filenames.iloc[i].filename    \n",
    "    list_filename_assembly.append(filename)\n",
    "\n",
    "    f=f\"/home/rebeca/Escritorio/Data/Assembly/{filename}\"\n",
    "    file = open(f,'r')\n",
    "    \n",
    "    split_contig_count=0\n",
    "    split_contig_N50=0\n",
    "    split_total_gap_length=0\n",
    "    split_gc_perc=0\n",
    "    while True:\n",
    "        content=file.readline()\n",
    "        pos=content.find('# Assembly name:')\n",
    "        if pos==0:\n",
    "            split=content.split(\"# Assembly name:\",1)[1]\n",
    "            split = split.replace(\" \", \"\")\n",
    "            list_assembly_name.append(split)\n",
    "        \n",
    "        pos=content.find('# RefSeq assembly accession:')\n",
    "        if pos==0:\n",
    "            split=content.split(\"# RefSeq assembly accession:\",1)[1]\n",
    "            split = split.replace(\" \", \"\")\n",
    "            list_refseq_assembly_accession.append(split)\n",
    "             \n",
    "        pos=content.find('all\tall\tall\tall\tspanned-gaps')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tspanned-gaps\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            list_spanned_gaps.append(split)\n",
    "\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tunspanned-gaps')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tunspanned-gaps\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            list_unspanned_gaps.append(split)\n",
    "\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tscaffold-count')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tscaffold-count\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            list_scaffold_count.append(split)\n",
    "\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tscaffold-N50')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tscaffold-N50\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            list_scaffold_N50.append(split)\n",
    "\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tcontig-count')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tcontig-count\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            split_contig_count=split\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tcontig-N50')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tcontig-N50\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            split_contig_N50=split\n",
    "            \n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\ttotal-gap-length')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\ttotal-gap-length\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            split_total_gap_length=split\n",
    "\n",
    "        \n",
    "        pos=content.find('all\tall\tall\tall\tgc-perc')\n",
    "        if pos==0:\n",
    "            split=content.split(\"all\tall\tall\tall\tgc-perc\",1)[1]\n",
    "            split = split.replace(\"\t\", \"\")\n",
    "            split_gc_perc=split\n",
    "            \n",
    "        if not content:\n",
    "            break\n",
    "\n",
    "    file.close()\n",
    "    list_contig_count.append(split_contig_count)\n",
    "    list_contig_N50.append(split_contig_N50)\n",
    "    list_total_gap_length.append(split_total_gap_length)\n",
    "    list_gc_perc.append(split_gc_perc)\n",
    "        \n",
    "    \n",
    "data_assembly = pd.DataFrame()\n",
    "data_assembly = pd.DataFrame(data_assembly,columns=['filename_assembly','assembly_name','refseq_assembly_accession','spanned_gaps','unspanned_gaps','scaffold_count','scaffold_N50','contig_count','contig_N50','total_gap_length','gc_perc'\n",
    "])\n",
    "\n",
    "data_assembly.filename_assembly=list_filename_assembly\n",
    "data_assembly.assembly_name=list_assembly_name\n",
    "data_assembly.refseq_assembly_accession=list_refseq_assembly_accession\n",
    "data_assembly.spanned_gaps=list_spanned_gaps\n",
    "data_assembly.unspanned_gaps=list_unspanned_gaps\n",
    "data_assembly.scaffold_count=list_scaffold_count\n",
    "data_assembly.scaffold_N50=list_scaffold_N50\n",
    "data_assembly.contig_count=list_contig_count\n",
    "data_assembly.contig_N50=list_contig_N50\n",
    "data_assembly.total_gap_length=list_total_gap_length\n",
    "data_assembly.gc_perc=list_gc_perc\n",
    "\n",
    "data_assembly.to_csv('Metadata_Assembly.csv', sep='\\t')\n",
    "data_assembly"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "674e109b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "8820edce",
   "metadata": {},
   "source": [
    "it=list(range(0, len(list_filenames.xml)))\n",
    "\n",
    "for i in it:\n",
    "    filename=list_filenames.xml.iloc[i]\n",
    "    print(filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5f71563a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "3198fb4c",
   "metadata": {},
   "source": [
    "### "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a423ac4e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "351da717",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d71527f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4708ed63",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcf1971d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d1ec000e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db7898d5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6359f4bf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35f118de",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75c88bfc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "729f2f8d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ba760d2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcb7d491",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "5f2a3df1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "06f35d0d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b44410f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7ba5ecc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b42e93c8",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0575ab76",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7584134a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1bec812f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fd0355a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "179c549d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71877a0e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3186731a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ad1ccadc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8299ae73",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48ccb8cd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "088a7803",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fc207372",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b010c3e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d5e4e69",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d99293a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "37d066a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "0e736399",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d753288c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6290680",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "483c19cc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ca06ab3",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "717451ec",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1fd3c0b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "695d9585",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f997ec42",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "534e7542",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ff59569f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "96b9f059",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f6aa414b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6a6b28a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
